src/java/org/apache/cassandra/cql3/PasswordObfuscator.java

```diff
@@ -35,6 +46,25 @@ public String obfuscate(String sourceString)
         if (passwordTokenStartIndex < 0)
             return sourceString;
 
-        return sourceString.substring(0, passwordTokenStartIndex + PASSWORD_TOKEN.length()) + OBFUSCATION_TOKEN;
+        return sourceString.substring(0, passwordTokenStartIndex + PASSWORD_TOKEN.length()) + " " + OBFUSCATION_TOKEN;
+    }
+
+    /**
+     * Obfuscates the passwords in a query
+     * 
+     * @param query The query whose passwords to obfuscate
+     * @param opts The options containing the password to obfuscate
+     * @return The query with obfuscated passwords
+     */
+    public static String obfuscate(String query, RoleOptions opts)
+    {
+        if (opts == null || query == null || query.isEmpty())
+            return query;
+
+        Optional<String> pass = opts.getPassword();
+        if (!pass.isPresent() || pass.get().isEmpty())
+            return query;
+        else
+            return query.replaceAll(pass.get(), PasswordObfuscator.OBFUSCATION_TOKEN);
```

Comment:
Using `replaceAll` does not look safe to me. If the password is `role` for example we will end up with a wrongly obfuscated statement that can reveal the actual password.

Reply:
Valid point and also if we commit this, it will be a regression to the current way of logging category DCL.

Reply:
This is a corner case. We can look if there is a way to avoid it. But I disagree it is a regression. The problem in GA atm is that obscuring queries it's a security problem imo from what I remember from OWASP. It disables the possibility of forensic analysis of logs upon a breach. Attackers use it to hide their steps.

So I see it as an improvement bc only non-parsing queries will suffer that.

Reply:
My point was simply that we can do something better than a `replaceAll`. We can find the precise location of the password based on the  `PASSWORD_TOKEN` and simply replace it.

Reply:
Aha good point. I will look into that thx.

Reply:
In the latest push I have attached an improved regex version of replaceAll that should do the trick imo as `PASSWORD` is a reserved word. if you take into account spaces, newlines, `=` present or not, quotes etc doing it without a regexp seems very complicated. wdyt?

Reply:
Tested this one with a few corner cases:

Type: audit
`LogMessage: user:cassandra|host:localhost/127.0.0.1:7000|source:/127.0.0.1|port:60355|timestamp:1636402050002|type:CREATE_ROLE|category:DCL|operation:create role 'superuser' with password = '*******' and superuser = true and login = true;`
We can probably add it as a unit test to cover it and be as a reminder?

Reply:
^what is the original cql you wanted added as a junit as I can only see the obscured version please?